### ISSUE TITLE
Gitignore file is not copied into application template

### ISSUE THREAD

**Issue by TimvdLippe** [Polymer/polymer-cli#531]
In #393 I submitted a PR to add `.gitignore` in the generator templates. However, today running with CLI version `0.18.0-alpha.8` no `.gitignore` was generated in the template.

**Comment by FredKSchott** [comment]
Was a new release stamped with this change? The CLI always uses the latest tagged release

**Comment by TimvdLippe** [comment]
According to https://github.com/Polymer/polymer-cli/commit/0bf0b4571acdea83397307937366e4dc9c6574f8 it shipped with version 0.17.0

**Comment by FredKSchott** [comment]
Oh right, I thought you meant starter-kit, shop, etc. I wonder if yeoman is ignoring dotfiles entirely then.

**Comment by TimvdLippe** [comment]
I think it is [this rule](https://github.com/Polymer/polymer-cli/blob/master/src/init/application/application.ts#L68) that is incorrect, but the regex syntax did not make sense to me with the `!`

**Comment by Borjagodoy** [comment]
I resolve this problem in my component generator with yeoman adding this excepcional copy for .gitignore we have changed the file .gitignore in the template folder to _gitignore because yeoman ignore the hidden file.

I can send a PR with them

`    this.fs.copy(
      this.templatePath('_gitignore'),
      this.destinationPath('.gitignore')
    );`

**Comment by Zecat** [comment]
In my generator, I was able to copy the hidden files by replacing
```js
// this
`${this.templatePath()}/**/?(.)!(_)*`
// to
`${this.templatePath()}/**/!(_)?(.)*`
```
and I guess
```js
// this could be replaced
this.fs.copyTpl(
  `${this.templatePath()}/**/?(.)!(_)*`,
  this.destinationPath(),
  this.props);

// with
this.fs.copyTpl(
  this.templatePath('**/!(_)?(.)*'),
  this.destinationPath());
```

Refers to https://github.com/Polymer/polymer-cli/blob/master/src/init/element/element.ts#L82

**Comment by Westbrook** [comment]
Dot files are touch, but `.gitignore` is especially tough, and I've used this syntax to get over this issue:
```
this.fs.copyTpl(
      `${this.templatePath()}/**/.!(gitignore)*`,
      this.destinationRoot(),
      this.props
    );

    this.fs.copyTpl(
      this.templatePath('.gitignorefile'),
      this.destinationPath(`.gitignore`),
      this.props
    );
```

**Comment by TimvdLippe** [comment]
Fixed in #957
